apps/veyfi/utils/index.ts

```diff
@@ -16,3 +17,39 @@ export function getVotingPower(lockAmount: BigNumber, unlockTime: TMilliseconds)
 	}
 	return lockAmount.div(MAX_LOCK).mul(duration);
 }
+
+export const keyBy = <T1, T2 extends keyof T1 & string>(array: T1[], key: T2): TDict<T1 | undefined> => 
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	(array || []).reduce((r, x): TDict<T1> => ({...r, [(x as any)[key]]: x}), {});
+
+export const sort = <T>(data: T[], by: Extract<keyof T, string>, order?: 'asc' | 'desc'): T[] => {
+	const compare = (a: T, b: T): number => {
+		const elementA = a[by];
+		const elementB = b[by];
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		if (!isNaN(elementA as any) && !isNaN(elementB as any)) {
```

Comment:
```suggestion
		if (!isNaN(elementA as number) && !isNaN(elementB as number)) {
```

Reply:
I think even better, how about this:
```typescript
		if (isNumber(elementA) && isNumber(elementB)) {
			return order === 'desc' ? elementA - elementB : elementB - elementA;
		}
```
and then have a type guard that guarantees the value is a number, like:
```typescript
function isNumber(value: unknown): value is number {
	return typeof value === 'number';
}
```

Reply:
I don't understand your refactor, with the function above we guarantee that both elements in the snippet below are numbers and we won't need to cast the strings, because we are sure they are numbers. That's the point of the type guard in my comment above, to tell TypeScript that we know those elements are numbers
```typescript
    if (isNumber(elementA) && isNumber(elementB)) {
        return order === 'desc' ? Number(elementA) - Number(elementB) : Number(elementB) - Number(elementA);
}
```

Reply:
You're totally right, should have checked the documentation for that [isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) 😅